lib/PuppeteerSharp/NetworkEventManager.cs

```diff
@@ -16,7 +16,8 @@ internal class NetworkEventManager
         private readonly ConcurrentDictionary<string, List<ResponseReceivedExtraInfoResponse>> _responseReceivedExtraInfoMap = new();
 
         public int NumRequestsInProgress
-            => _httpRequestsMap.Values.Count(r => r.Response == null);
+            => _httpRequestsMap.Values.Count(r => r.Response == null
+                                                  && string.IsNullOrWhiteSpace(r.Failure));
```

Comment:
@kblok I did this fix after debag ;)

I'm doing some SSR for crawlers.

`               
                await using var browser = await GetBrowserAsync();
                await using var page = await browser.NewPageAsync();
                await page.SetRequestInterceptionAsync(true);
                page.Request += async (sender, args) =>
                {
                    if (args.Request.Url.Contains("googletagmanager", StringComparison.InvariantCultureIgnoreCase)
                         || args.Request.Url.Contains("amplitude", StringComparison.InvariantCultureIgnoreCase)
                         || args.Request.Url.Contains("intercom.io", StringComparison.InvariantCultureIgnoreCase)
                         || args.Request.Url.Contains("google-analytics", StringComparison.InvariantCultureIgnoreCase))
                        await args.Request.AbortAsync();
                    else
                        await args.Request.ContinueAsync();
                };
                await page.GoToAsync(url);
                await page.WaitForNetworkIdleAsync(new WaitForNetworkIdleOptions { IdleTime = 1000 });
                var content = await page.GetContentAsync();
                return content;
                `

WaitForNetworkIdleAsync - will throw timeout exception (because some part of requests aborted)
Logic of this method waiting when counter will be 0, but failed requests included into NumRequestsInProgress.

Url: https://defi.watch/tokens/wbnb